server/lib/ga-collector.js

```diff
@@ -103,6 +107,31 @@ GACollector.prototype = {
         });
       }
     });
+  },
+  /**
+   * Calculate the queue time of the event.
+   * See https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters?hl=en#qt
+   *
+   * @param {Number} startTime
+   *  The time from which all offsets are calculated in the front-end metrics.
+   * @param {Number} flushTime
+   *  The time when the events were flushed and sent to the server in the front-end metrics.
+   * @param {Number} offset
+   *  Offset of the event since metrics start
+   *
+   * @returns {Number}
+   *  The time delta (in milliseconds) between when the hit being reported occurred and the time the hit was sent.
+   *
+   * @private
+   */
+  _calculateQueueTime: function (startTime, flushTime, offset) {
+    if (! startTime || ! flushTime || ! offset) {
+      return 0;
+    }
+
+    var offsetTime = startTime + offset;
+
+    return flushTime - offsetTime;
```

Comment:
Should we ensure the returned value is always >= 0, just in case? Maybe wrap the calculation in a `Math.max(0, ...`?